### PR TITLE
Adds functionality to specify additional groups to join.

### PR DESCRIPTION
**Pull request by mrunalp**
A stab at https://github.com/docker/docker/issues/9360

Signed-off-by: Mrunal Patel <mrunalp@gmail.com> (github: mrunalp)

### CODE REVIEW

**Comment by vishh** [comment]
LGTM

**Comment by LK4D4** [comment]
LGTM

**Comment by thockin** [comment]
LGTM

File: namespaces/init.go [hunk 1]
```diff
@@ -188,12 +188,17 @@ func SetupUser(u string) error {
...
 	}
 
-	execUser, err := user.GetExecUserPath(u, &defaultExecUser, passwdPath, groupPath)
+	execUser, err := user.GetExecUserPath(container.User, &defaultExecUser, passwdPath, groupPath)
 	if err != nil {
 		return fmt.Errorf("get supplementary groups %s", err)
 	}
 
-	if err := syscall.Setgroups(execUser.Sgids); err != nil {
+	suppGroups := execUser.Sgids
```
Comment (vmarmol) [inline namespaces/init.go]: I think this can just be append(execUser.Sgids, container.AdditionalGroups...) since empty lists are handled.

**Comment by vmarmol** [comment]
Outside of that, LGTM

File: namespaces/init.go (hunk 1 above)
Comment (mrunalp) [inline namespaces/init.go]: Fixed, thanks.

**Comment by vmarmol** [comment]
LGTM

**Pull request merged**